### PR TITLE
Add first adapter to third-party tool

### PR DESCRIPTION
**Pull request by bachand**
Best reviewed commit by commit.

This repo will be more useful if there is an easy way to adapt the Airbnb YAML file to third-party linting tools. Here we add an adapter to the [Woke](https://github.com/get-woke/woke) tool.

@airbnb/inclusion-repo-maintainers 
cc: @caitlinelfring who may be interested

### CODE REVIEW

File: adapters/generate_woke_config [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
```
Comment (dfed) [inline adapters/generate_woke_config:1]: should we rename this file to have a `.rb` suffix?

File: README.md [hunk 2]
```diff
@@ -8,5 +8,9 @@ We are publicly sharing `infractions.yml`. `infractions.yml` is a data file of t
 
 The data file format is agnostic to any particular tool and is meant to be general enough to be adapted to any tool.
 
+## Adapters
+
+The `adapters` folder includes shell scripts to adapt the Airbnb infractions YAML file to third-party tools. Airbnb welcomes contributions for new tools and systems that promote belonging. An adapter for a third-party tool does not necesarily mean that Airbnb endorses the third-party tool.
```
Comment (dfed) [inline README.md:13]: How do we review changes to adapters? Or put another way: how will we have the context to review changes to adaptors we don't endorse or know well?

File: Rakefile [hunk 3]
```diff
@@ -18,3 +18,13 @@ task :validate_data do
...
 end
+
+# Ensures that the script to generate a woke config file produces valid YAML.
+task :validate_woke_adapter do
+  require 'yaml'
+  infractions_file = File.join(__dir__, 'infractions.yml')
+  generate_script = File.join(__dir__, 'adapters/generate_woke_config')
+  output = `#{generate_script} #{infractions_file}`
+  woke_config = YAML.load(output)
+  raise 'Woke config contains no entries' unless woke_config['rules'].length > 1
```
Comment (dfed) [inline Rakefile:29]: why are we checking `> 1` here? I'd have guessed that we'd want `> 0`, but I'm likely missing something.

**Review by dfed — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (bachand) [reply README.md:13]: This is a good point. Let's remove the general concept of adapters until the need arises.

File: Rakefile (hunk 3 above)
Comment (bachand) [reply Rakefile]: Good point 15aee0d

File: adapters/generate_woke_config (hunk 1 above)
Comment (bachand) [reply adapters/generate_woke_config:1]: With the shebang we are telling the shell how to execute this script. It's idiomatic in my experience to leave off the extension for standalone scripts/executables like this under the reasoning that the user doesn't need to know what language it is. They simply need to know it's executable (which we've indicated with the file permissions). In this way the implementation of the tool could change (e.g. Ruby to Python) and the user wouldn't need to change any code that executes the script.

File: README.md (hunk 2 above)
Comment (bachand) [reply README.md]: https://github.com/airbnb/inclusion/pull/16/commits/37be8215801ef2d790a4113088416564a8120b28

**Comment by bachand** [comment]
@dfed lmk what you think of the changes. Thanks for the thorough review.

**Review by dfed — APPROVED** [review]
dig it

**Comment by caitlinelfring** [comment]
Hey @bachand! Just remembering this PR now... I'm curious if Airbnb is using woke and if so, would there be any interest in including Airbnb in the "who uses woke" section of the readme? https://github.com/get-woke/woke#who-uses-woke. Happy to discuss offline if you prefer!

**Comment by bachand** [comment]
Hey @caitlinelfring ! Maybe we can find a time to Zoom? I'd love to tell you about the work we're doing at Airbnb and discuss how we can collaborate. What is a good way to contact you? Alternatively, you may be able to guess my email address at airbnb.com 😄

**Pull request merged**